profiler/src/main/java/com/splunk/opentelemetry/profiler/ThreadDumpToStacks.java

```diff
@@ -16,84 +16,50 @@
+      }
+    }
 
-  private final StackTraceFilter stackTraceFilter;
-
-  public ThreadDumpToStacks(StackTraceFilter stackTraceFilter) {
-    this.stackTraceFilter = stackTraceFilter;
-  }
-
-  public Stream<String> toStream(String wallOfStacks) {
-    Spliterator<String> spliterator = new StackSpliterator(wallOfStacks, stackTraceFilter);
-    return StreamSupport.stream(spliterator, false);
+    return false;
   }
 
-  private static class StackSpliterator extends Spliterators.AbstractSpliterator<String> {
-    private final String wallOfStacks;
-    private final StackTraceFilter stackTraceFilter;
-    private int start = 0;
-    private int next = 0;
-    private boolean done = false;
+  private boolean findNextCandidate(ThreadDumpRegion region) {
+    String threadDump = region.threadDump;
+    int start = region.endIndex;
 
-    public StackSpliterator(String wallOfStacks, StackTraceFilter stackTraceFilter) {
-      super(Long.MAX_VALUE, Spliterator.IMMUTABLE | Spliterator.ORDERED);
-      this.wallOfStacks = wallOfStacks;
-      this.stackTraceFilter = stackTraceFilter;
-    }
-
-    @Override
-    public boolean tryAdvance(Consumer<? super String> action) {
-      while (true) {
-        if (done) {
-          return false;
-        }
-        next = wallOfStacks.indexOf("\n\n", start);
-        if (next == -1) {
-          next = wallOfStacks.lastIndexOf('\n', start);
-        }
-        if (next == -1) {
-          done = true;
-          return false;
-        }
-        // Reached the end of the wall, so just set next to the end
-        if (next < start) {
-          next = wallOfStacks.length() - 1;
-        }
-        if (stackTraceFilter.test(wallOfStacks, start, next)) {
-          action.accept(advanceNextStack());
-          return true;
-        }
-        start = next + 1;
-        while ((start < wallOfStacks.length()) && (wallOfStacks.charAt(start) == '\n')) {
-          start++;
-        }
-        if (start >= wallOfStacks.length()) {
-          done = true;
-          return false;
-        }
+    // skip over any newlines, returning failure in case we reach end of string this way
+    while (true) {
+      if (start >= threadDump.length()) {
+        return false;
+      } else if (threadDump.charAt(start) != '\n') {
+        break;
       }
+      start++;
     }
 
-    private String advanceNextStack() {
-      String result = wallOfStacks.substring(start, next);
-      if (next >= wallOfStacks.length() - 2) {
-        done = true;
-      }
-      start = next + 2; // next chunk starts after the "\n\n"
-      next = start;
-      return result;
+    int end = threadDump.indexOf("\n\n", start);
+    if (end == -1) {
+      end = threadDump.lastIndexOf('\n', start);
     }
+    if (end == -1) {
+      return false;
+    }
+    // Reached the end of the wall, so just set next to the end
+    if (end < start) {
+      end = region.threadDump.length();
+    }
+
+    region.startIndex = start;
+    region.endIndex = end;
```

Comment:
WDYT about encapsulating this modification in a `move(start, end)` method?

Reply:
Irrelevant now as `ThreadDumpToStacks` migrated to a method within `ThreadDumpRegion`.